internal/datasourcev2/polling_http_request.go

```diff
@@ -72,13 +72,10 @@ func (r *pollingRequester) Request() (*fdv2proto.ChangeSet, error) {
 		r.loggers.Debug("Polling LaunchDarkly for feature flag updates")
 	}
 
-	body, cached, err := r.makeRequest(endpoints.PollingRequestPath)
```

Comment:
This is a hack because our current data sources are started/stopped rather than being re-instantiated. The issue is - if we had an error response (like the payload was malformed or just an HTTP error), we'd get back an empty changeset (`.NoChanges()`) with no error (`nil`).

This means if the previous Data Source Status was something like `VALID` due to a previous synchronizer, then we start up polling and it gets the same response as it did last time, we wouldn't update the state to `INTERRUPTED` based on this error.

If instead the data source was re-instantiated, then there would be no "previous state" for this new run of the data source.